diktat-maven-plugin/src/main/kotlin/com/saveourtool/diktat/plugin/maven/DiktatBaseMojo.kt

```diff
@@ -160,7 +160,30 @@ abstract class DiktatBaseMojo : AbstractMojo() {
         return generateSequence(mavenProject) { it.parent }
             .map { it.basedir.toPath().resolve(diktatConfigFile) }
             .run {
-                firstOrNull { it.isRegularFile() } ?: first()
+                firstOrNull { it.isRegularFile() } ?: file
             }
     }
+
+    private fun files(): List<Path> {
+        val (excludedDirs, excludedFiles) = excludes.map(::File).partition { it.isDirectory }
+        return inputs
+            .asSequence()
+            .map(::File)
+            .flatMap {
+                it.files(excludedDirs, excludedFiles)
+            }
+            .map { it.toPath() }
+            .toList()
+    }
+
+    @Suppress("TYPE_ALIAS")
+    private fun File.files(
+        excludedDirs: List<File>,
+        excludedFiles: List<File>,
+    ): Sequence<File> = walk()
+        .filter { file ->
+            file.isDirectory || file.toPath().isKotlinCodeOrScript()
```

Comment:
why we need first to filter as `file.isDirectory` and then `.filter { it.isFile }`?

Reply:
just copied from old PR #1655. Will look again

Reply:
refactored